board-games/board-games/GameOfLife_MainWindow.xaml.cs

```diff
@@ -0,0 +1,104 @@
+        /// </exception>
+        private List<int> GenerateRandomIndices(int numberOfIndices)
+        {
+            if (numberOfIndices > NumberOfTiles)
+            {
+                throw new ArgumentOutOfRangeException("The number of randomly generated indices cannot be larger than the total number of tiles.");
+            }
+            if (numberOfIndices < 0)
+            {
+                throw new ArgumentOutOfRangeException("The number of randomly generated indices cannot be smaller than 0");
+            }
+            var lowerBound = 1;
+            var upperBound = NumberOfTiles + 1;
+            List<int> randomNumbersList = new List<int>();
+            Random random = new Random();
+
+            while (numberOfIndices > 0)
+            {
+                var randomNumber = random.Next(lowerBound, upperBound);
+
+                if (!randomNumbersList.Contains(randomNumber))
+                {
+                    randomNumbersList.Add(randomNumber);
+                    numberOfIndices--;
+                }
+            }
+            return randomNumbersList;
+        }
+        private void MainWindow_Loaded(object sender, RoutedEventArgs e)
+        {
+            int nextIndex = 0;
+            List<int> indicesToColor = GenerateRandomIndices(TotalNumberOfEventTiles + NumberOfGreenTiles);
+            List<int> indicesOfGreenTiles = indicesToColor.GetRange(nextIndex, NumberOfGreenTiles);
+            // ff96bb20
+            ColorTiles(indicesOfGreenTiles, Color.FromRgb(150,187,32));
+            nextIndex = NumberOfGreenTiles;
+            List<int> indicesOfRedTiles = indicesToColor.GetRange(nextIndex, NumberOfEventTilesByCategory);
+            ColorTiles(indicesOfRedTiles, Color.FromRgb(229, 51, 61));
+            nextIndex = nextIndex + NumberOfEventTilesByCategory;
+            List<int> indicesOfBlueTiles = indicesToColor.GetRange(nextIndex, NumberOfEventTilesByCategory);
+            ColorTiles(indicesOfBlueTiles, Color.FromRgb(43, 163, 171));
+            nextIndex = nextIndex + NumberOfEventTilesByCategory;
+            List<int> indicesOfOrangeTiles = indicesToColor.GetRange(nextIndex, NumberOfEventTilesByCategory);
+            ColorTiles(indicesOfOrangeTiles, Color.FromRgb(241, 151, 54));
+        }
+        /// <summary>
+        /// Colors the tiles with specified indices using the given brush color.
+        /// </summary>
+        /// <param name="tileIndexesToColor">The indices of the tiles to be colored.</param>
+        /// <param name="givenBrushColor">The color to fill the tiles with.</param>
+        private void ColorTiles(List<int> tileIndexesToColor, Color givenBrushColor)
+        {
+            SolidColorBrush currentBrush = new SolidColorBrush(givenBrushColor);
+            
+            string currentTileName = "Tile";
+            foreach(int tileIndex in tileIndexesToColor)
+            {
+                currentTileName = currentTileName + tileIndex.ToString();
```

Comment:
If we had a constant ConstantName="Tile", we could have here: string currentTileName = ConstantName + ...
Also, in this case the last line in the foreach will not be necessary.